doc/api/test.md

```diff
@@ -454,6 +451,166 @@ test('spies on an object method', (t) => {
+      case 'test:plan':
+        callback(null, 'test plan');
+        break;
+      case 'test:diagnostic':
+        callback(null, event.data.message);
+        break;
+    }
+  },
+});
+
+export default customReporter;
+```
+
+```cjs
+const { Transform } = require('node:stream');
+
+const customReporter = new Transform({
+  writableObjectMode: true,
+  transform(event, encoding, callback) {
+    switch (event.type) {
+      case 'test:start':
+        callback(null, `test ${event.data.name} started`);
+        break;
+      case 'test:pass':
+        callback(null, `test ${event.data.name} passed`);
+        break;
+      case 'test:fail':
+        callback(null, `test ${event.data.name} failed`);
+        break;
+      case 'test:plan':
+        callback(null, 'test plan');
+        break;
+      case 'test:diagnostic':
+        callback(null, event.data.message);
+        break;
+    }
+  },
+});
+
+module.exports = customReporter;
+```
+
+Example of a custom reporter using a generator function:
+
+```mjs
+export default async function * customReporter(source) {
+  for await (const event of source) {
+    switch (event.type) {
+      case 'test:start':
+        yield `test ${event.data.name} started\n`;
+        break;
+      case 'test:pass':
+        yield `test ${event.data.name} passed\n`;
+        break;
+      case 'test:fail':
+        yield `test ${event.data.name} failed\n`;
+        break;
+      case 'test:plan':
+        yield 'test plan';
+        break;
+      case 'test:diagnostic':
+        yield `${event.data.message}\n`;
+        break;
+    }
+  }
+}
+```
+
+```cjs
+module.exports = async function * customReporter(source) {
+  for await (const event of source) {
+    switch (event.type) {
+      case 'test:start':
+        yield `test ${event.data.name} started\n`;
+        break;
+      case 'test:pass':
+        yield `test ${event.data.name} passed\n`;
+        break;
+      case 'test:fail':
+        yield `test ${event.data.name} failed\n`;
+        break;
+      case 'test:plan':
+        yield 'test plan\n';
+        break;
+      case 'test:diagnostic':
+        yield `${event.data.message}\n`;
+        break;
+    }
+  }
+};
+```
+
+### Multiple reporters
+
+The [`--test-reporter`][] flag can be specified multiple times to report test
+results in several formats. In this situation
+it is required to specify a destination for each reporter
+using [`--test-reporter-destination`][].
+Destination can be `stdout`, `stderr`, or a file path.
+Reporters and destinations are paired according
+to the order they were specified.
+
+In the following example, the `spec` reporter will output to `stdout`,
+and the `dot` reporter will output to `file.txt`:
+
+```bash
+node --test-reporter=spec --test-reporter=dot --test-reporter-destination=stdout --test-reporter-destination=file.txt
+```
+
+When a single reporter is specified, or the default reporter is used,
+the destination , unless a destination is explicitly provided.
```

Comment:
This line doesn't make sense to me now.

Also, my original comment in https://github.com/nodejs/node/pull/45712#discussion_r1051448739 doesn't seem to be valid. If you don't explicitly specify a reporter, you cannot specify a destination for the default reporter.

Reply:
fixed the docs. preferred not to change the behavior since TAP is the default reporter and it is only valid on `stdout`